test/saithrift/tests/sail3.py

```diff
@@ -1477,3 +1477,192 @@ def runTest(self):
+                                            eth_src=src_mac,
+                                            ip_src=ip_src,
+                                            ip_dst=ip_dst,
+                                            ip_id=i,
+                                            tcp_sport=sport,
+                                            tcp_dport=dport,
+                                            ip_ttl=64)
+                    exp_pkt = simple_tcp_packet(
+                                            eth_dst=self.mac_pool[0],
+                                            eth_src=router_mac,
+                                            ip_src=ip_src,
+                                            ip_dst=ip_dst,
+                                            ip_id=i,
+                                            tcp_sport=sport,
+                                            tcp_dport=dport,
+                                            ip_ttl=63)
+                    masked_exp_pkt = Mask(exp_pkt)
+                    masked_exp_pkt.set_do_not_care_scapy(ptf.packet.Ether,"dst")
+
+                    send_packet(self, 0, str(pkt))
+                    (match_index,rcv_pkt) = verify_packet_any_port(self,masked_exp_pkt,destanation_ports)
+                    logging.debug("found expected packet from port %d" % destanation_ports[match_index])
+                    pkt_counter[match_index] += 1
+                    sport = random.randint(0,0xffff)
+                    dport = random.randint(0,0xffff)
+                        
+        #final uniform distribution check
+        logging.debug(pkt_counter)
+        logging.debug(first_rif_port)
+        lag_packets = sum(pkt_counter[1:first_rif_port])
+        lag_average = lag_packets/(len(self.lag_members) + 1)
+        logging.debug("the sum of packets through the lag is " + str(lag_packets))
+        logging.debug("the lag average for the lag is " + str(lag_average))
+        for stat_port in range(1,first_rif_port):
+            logging.debug( "PORT #"+str(stat_port)+":")
+            logging.debug(str(pkt_counter[stat_port]))
+            self.polarizationCheck(pkt_counter[stat_port],lag_average)
+        rifs_average = sum(pkt_counter)/(len(self.port_rifs) + 1)
+        logging.debug("lag average " + str(lag_average))
+        self.polarizationCheck(lag_packets,rifs_average)
+        for stat_port in range(first_rif_port,self.total_changing_ports):
+            logging.debug( "PORT #"+str(stat_port)+":")
+            logging.debug(str(pkt_counter[stat_port]))
+            self.polarizationCheck(pkt_counter[stat_port],rifs_average)
+    
+                 
+    def runTest(self):
+        """
+        For sai server, testing different lags with router
+        ---- Test for 16 ports minimun ----
+        Steps
+        1. Create virtual router, lag with rif and rif for src port
+        2. add ports to the lag and the rest of the ports connect to rifs
```

Comment:
I'd like you to move "create lag with rif" into step 2.

In step 2, aka, setup_rifs, you create the ecmp group, create the lag, add ports to lag and create port rifs.

In setup_rifs, it includes step 2, 3, and 4.